beacon_node/lighthouse_network/src/service/mod.rs

```diff
@@ -332,17 +356,9 @@ impl<E: EthSpec> Network<E> {
             // If we are using metrics, then register which topics we want to make sure to keep
             // track of
             if ctx.libp2p_registry.is_some() {
-                let topics_to_keep_metrics_for = attestation_sync_committee_topics::<E>()
-                    .map(|gossip_kind| {
-                        Topic::from(GossipTopic::new(
-                            gossip_kind,
-                            GossipEncoding::default(),
-                            enr_fork_id.fork_digest,
-                        ))
-                        .into()
-                    })
-                    .collect::<Vec<TopicHash>>();
-                gossipsub.register_topics_for_metrics(topics_to_keep_metrics_for);
+                for topics in max_topics_by_fork {
+                    gossipsub.register_topics_for_metrics(topics);
+                }
```

Comment:
why do we want to register metric for all current + future topics on startup? 

This takes up unnecessary extra storage for metric databases and I don't think this actually works because there's a limit on how many unsubscribed topics you can register metric for:
https://github.com/sigp/lighthouse/blob/e7ea69647a4cb686d0c7a80f24b3f34d329a7e01/beacon_node/lighthouse_network/gossipsub/src/metrics.rs#L39-L41

Reply:
Ignore my comment about storage - i thought this was proemtheus metrics, but I think the comment on max still applies.

Reply:
Sorry looks like I misunderstood how this works, this function just register it to make it an allowed topic.
And the fork digest label only gets created _if_ the metric is updated for the first time, so we should be good storage wise.

Do you know the reason for only allowing sync committee topics previously?

Reply:
Checked with Age and this is fine!